quick-sight.tf

```diff
@@ -0,0 +1,59 @@
+# #########################################
+# QUICK SIGHT
+# #########################################
+resource "aws_security_group" "quick_sight_sg" {
+  count       = var.enable_quick_sight ? 1 : 0
+  name        = "${module.labels.id}-quick-sight"
+  vpc_id      = module.vpc.vpc_id
+  description = lower("${module.labels.id}-quick-sight-service security group")
+  tags        = module.labels.tags
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_security_group_rule" "quick_sight_ingress" {
+  count                    = var.enable_quick_sight ? 1 : 0
+  description              = "Allows Quick Sight Connections"
+  type                     = "ingress"
+  from_port                = 5432
+  to_port                  = 5432
+  protocol                 = "tcp"
+  source_security_group_id = aws_security_group.quick_sight_service_sg[0].id
+  security_group_id        = aws_security_group.quick_sight_sg[0].id
+}
+
+resource "aws_security_group" "quick_sight_service_sg" {
+  count       = var.enable_quick_sight ? 1 : 0
+  name        = "${module.labels.id}-quick-sight-service"
+  vpc_id      = module.vpc.vpc_id
+  description = lower("${module.labels.id}-quick-sight-service security group")
+  tags        = module.labels.tags
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_security_group_rule" "quick_sight_service_ingress" {
```

Comment:
Thinking about this one ?
Does this open us up to widely

Reply:
I think it does but it also limits the traffic between 2 specific SGs, it seems ok to me.

Reply:
the only concern I have is the circular dependency added. I think we need to use `depends_on` clause on SG to specify in which orders rules needs to be applied/removed so that we can safely crate and destroy without ending in the problem we had during the `dev` phase of this feature